### PR TITLE
Allow Add Organization

### LINKED ISSUE #2839: Allow anyone to add an organization
# Description 
At the moment, only admins can create new organizations. For the APInf SaaS service, it must be possible for any user to create an organization as a default. (See a separate issue for a setting, which allows only admins to add organizations.)

# Goal
* [ ] Any logged-in user can see the Add organization button in Organization Catalog
* [ ] Any logged in user is allowed to create an organization

### LINKED ISSUE #2840: Add a setting for "Only admins can add organizations"
# Description
After task #2839 is completed, any logged-in user is able to create a new organizations. In some deployments, it maybe necessary to be able to configure this and allow only admins to create new organizations. As there is a setting for "Only platform administrators are allowed to add new APIs", there should be a setting for "Only platform administrators are allowed to add new organizations".

# Goal
* [ ] It is possible to configure in Settings, whether  any logged-in user or  only platform administrator(s) can add new organization(s)
* [ ] If the restriction "Only platform administrators are allowed to add new organizations" is applied
  * [ ]  the button "Add new organization is available only for admin uses
  * [ ] Only admins  are able to create organizations

Related to #2839

# Wireframe
image

### PR DESCRIPTION
**Pull request by marla-singer**
Closes #2839 
Closes #2840

### CODE REVIEW

File: apinf_packages/organizations/client/catalog/catalog.js [hunk 1]
```diff
@@ -121,6 +126,34 @@ Template.organizationCatalog.helpers({
...
+        // Otherwise check if current user is admin
+        return Roles.userIsInRole(userId, ['admin']);
+      }
+
+      // Return true because no settings are set
+      // By default allowing all user to add an Organization
+      return true;
+    }
+
+    // If user isn'tloggin then don't allow
```
Comment (brylie) [inline apinf_packages/organizations/client/catalog/catalog.js]: Fix typo.

File: apinf_packages/organizations/collection/permissions.js [hunk 2]
```diff
@@ -6,20 +6,34 @@ https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence
...
 Organizations.allow({
   insert (userId) {
-    // Check if current user is admin
-    const userIsAdmin = Roles.userIsInRole(userId, ['admin']);
+    // Get settigns document
+    const settings = Settings.findOne();
+
+    if (settings) {
+      // Get value of field or false as default value
+      const onlyAdmins = _.get(settings, 'access.onlyAdminsCanAddOrganizations', false);
```
Comment (brylie) [inline apinf_packages/organizations/collection/permissions.js]: Rename to `onlyAdminsCanAdd` or `onlyAdminsCanInsert`.

File: apinf_packages/organizations/collection/permissions.js [hunk 3]
```diff
@@ -6,20 +6,34 @@ https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence
...
+
+      // Allow user to add an Organization because not only for admin
```
Comment (brylie) [inline apinf_packages/organizations/collection/permissions.js]: Move comment inside of `if` block (above `return` statement).

File: apinf_packages/core/client/navbar/navbar.js [hunk 4]
```diff
@@ -27,8 +27,8 @@ Template.navbar.onCreated(function () {
   instance.autorun(() => {
```
[the remark this answers is not in the document]
Comment (marla-singer) [reply apinf_packages/core/client/navbar/navbar.js]: @brylie I don't see any reason for that

**Comment by marla-singer** [comment]
@brylie Fixed requested changes

**Comment by brylie** [comment]
Since we are touching the schema, should we consider adding a migration to this PR?

**Comment by marla-singer** [comment]
> Since we are touching the schema, should we consider adding a migration to this PR?

@brylie The field isn't required then I didn't add a migration

File: apinf_packages/core/client/navbar/navbar.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (brylie) [reply apinf_packages/core/client/navbar/navbar.js]: The two places where this subscription are used are:

- navbar
- organization catalog

What common 'parent' to the navbar and organization catalog templates share?

File: apinf_packages/core/client/navbar/navbar.js [hunk 5]
```diff
@@ -27,8 +27,8 @@ Template.navbar.onCreated(function () {
...
     // Check if user is logged in
     if (Meteor.userId()) {
-      // If user logged in, subscribe to 'onlyAdminCanAddApis' setting
-      instance.subscribe('singleSetting', 'access.onlyAdminsCanAddApis');
+      // If user logged in, subscribe to Access setting
+      instance.subscribe('singleSetting', 'access');
```
Comment (brylie) [inline apinf_packages/core/client/navbar/navbar.js]: Rename `instance` to `templateInstance`.

Comment (marla-singer) [reply apinf_packages/core/client/navbar/navbar.js]: Done

**Pull request merged**